### ISSUE TITLE
first impressions

### ISSUE THREAD

**Issue by nilsbecker** [nojb/llpp#2]
(disclaimer: i'm not a previous llpp user)

key shortcuts seem to be working as advertised, including ctrl+... .  good. this is not always the case for applications ported to os x.

no retina support: this is probably the underlying mupdf? having this would be a huge plus (put differently, not having it is a problem ;)

crazy amount of debugging info logged to the terminal -- but i guess this is normal for a work in progress.

applying `w` shortcut repeatedly shrinks the window more and more

no sideways scrolling with the trackpad

no pinch zoom

this is probably upstream: i was hoping to find a mode where one can set a viewport for a page (e.g. just enough to cover the actually used area of the page) and flip fwd/bwd in the document while conserving the viewport. the point is to maximize the available resolution. does this exist?

**Comment by nilsbecker** [comment]
(also: it feels very snappy)

**Comment by nojb** [comment]
>
key shortcuts seem to be working as advertised, including ctrl+... . good. this is not always the case for applications ported to os x.

>
no retina support: this is probably the underlying mupdf? having this would be a huge plus (put differently, not having it is a problem ;)

I think this is doable but it needs some support from the drawing code (it needs to render things at 4x the size).  Certainly supported by OS X OpenGL machinery.  /cc @moosotc

>
crazy amount of debugging info logged to the terminal -- but i guess this is normal for a work in progress.

Indeed, this will go away once things are working.

>
applying w shortcut repeatedly shrinks the window more and more

I will look into this.

>
no sideways scrolling with the trackpad

I think I know how to fix this, will try to do so soon.

>
no pinch zoom

I am not sure how this is implemented.  I will look into it.

>
this is probably upstream: i was hoping to find a mode where one can set a viewport for a page (e.g. just enough to cover the actually used area of the page) and flip fwd/bwd in the document while conserving the viewport. the point is to maximize the available resolution. does this exist?

I don't think so at the moment, but not really sure /cc @moosotc.

**Comment by moosotc** [comment]
Nicolas Ojeda Bar <notifications@github.com> writes:

>>
> key shortcuts seem to be working as advertised, including
> ctrl+... . good. this is not always the case for applications ported
> to os x.
>
>>
> no retina support: this is probably the underlying mupdf? having this
> would be a huge plus (put differently, not having it is a problem ;)

I only have access to mac book air and that does not have retina...

>
> I think this is doable but it needs some support from the drawing code
> (it needs to render things at 4x the size).  Certainly supported by OS
> X OpenGL machinery.  /cc @moosotc
>

Isn't this retina stuff opt-in so to speak? Perhaps just setting some
flag in cocoa initialization code will just make things work?

>>
> crazy amount of debugging info logged to the terminal -- but i guess
> this is normal for a work in progress.
>
> Indeed, this will go away once things are working.
>
>>
> applying w shortcut repeatedly shrinks the window more and more
>
> I will look into this.
>

Don't use 'w' it's an remnant of day long gone by, i should probably
remove it completely.

>>
> no sideways scrolling with the trackpad
>
> I think I know how to fix this, will try to do so soon.
>
>>
> no pinch zoom
>
> I am not sure how this is implemented.  I will look into it.
>
>>
> this is probably upstream: i was hoping to find a mode where one can
> set a viewport for a page (e.g. just enough to cover the actually used
> area of the page) and flip fwd/bwd in the document while conserving
> the viewport. the point is to maximize the available resolution. does
> this exist?
>

Try pressing 'P' followed by 'ctrl-3'. Not sure if that's what you want
though.

> I don't think so at the moment, but not really sure /cc @moosotc.
>
> ---
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/nojb/llpp/issues/2#issuecomment-231717426

-- 
mailto:moosotc@gmail.com

**Comment by nojb** [comment]
>
Isn't this retina stuff opt-in so to speak? Perhaps just setting some
flag in cocoa initialization code will just make things work?

Yes, but the OpenGL calls that work in pixels need to have their arguments multiplied by an appropiate factor (4 ?) IIUC.  See

https://developer.apple.com/library/mac/documentation/GraphicsAnimation/Conceptual/HighResolutionOSX/CapturingScreenContents/CapturingScreenContents.html

under the heading "Check for Calls Defined in Pixel Dimensions".

**Comment by nilsbecker** [comment]
> Try pressing 'P' followed by 'ctrl-3'. Not sure if that's what you want
> though.

thanks, i tried. this puts me in presentation mode, and sets the fit model to proportional (yes?) but then i do not see how to jump to the same zoom and port on the next page and back. j/k dont do it and space also does not...

**Comment by moosotc** [comment]
nilsbecker <notifications@github.com> writes:

>> Try pressing 'P' followed by 'ctrl-3'. Not sure if that's what you want
>> though.
>
> thanks, i tried. this puts me in presentation mode, and sets the fit
> model to proportional (yes?) but then i do not see how to jump to the
> same zoom and port on the next page and back. j/k dont do it and space
> also does not...

I'm afraid I do not quite get what is that you want exactly.

Perhaps you want 'page' fit model (one more ctrl-3, or you can set it
explicitly in the infor screen 'i')

[..snip..]

-- 
mailto:moosotc@gmail.com

**Comment by nilsbecker** [comment]
sorry, maybe i'm making it too complicated. here is the general problem: monitors often do not have enough resolution to read a pdf comfortably in full-page (not talking about retina now). and scrolling all the time is just so annoying. so i would like to zoom in as much as i can to use the resolution i have to the full extent. 

ie. i want to zoom to a viewport that includes all the printed text but not the white borders of the page or the page numbers. then i want to keep that viewport fixed and flip pages, reading the whole document without ever scrolling.

**Comment by moosotc** [comment]
nilsbecker <notifications@github.com> writes:

> sorry, maybe i'm making it too complicated. here is the general
> problem: monitors often do not have enough resolution to read a pdf
> comfortably in full-page (not talking about retina now). and scrolling
> all the time is just so annoying. so i would like to zoom in as much
> as i can to use the resolution i have to the full extent.
>
> ie. i want to zoom to a viewport that includes all the printed text
> but not the white borders of the page or the page numbers. then i want
> to keep that viewport fixed and flip pages, reading the whole document
> without ever scrolling.

Tried trimming the margins? '-T' or, again, checking the 'trim margins'
in the infor screen ('i')

-- 
mailto:moosotc@gmail.com

**Comment by nilsbecker** [comment]
yes! nice! i've been looking for this feature but never found a viewer that does it.

**Comment by nojb** [comment]
>
no retina support: this is probably the underlying mupdf? having this would be a huge plus (put differently, not having it is a problem ;)

Retina should now be supported.  @nilsbecker, could you give a try and let me know if it works for you ?

**Comment by nilsbecker** [comment]
i tried, it works! great. however, the help text accessible with `h` is now tiny, as well as the status line and i guess actually all ui text. that should be scaled up again -- it could still remain a bit smaller than in the non-retina version but at least twice the current size i think.

**Comment by nojb** [comment]
Great!  I will fix the font size issues tonight.

@moosotc:  what's the best way to multiply all font sizes by the given factor `Wsi.fontsizefactor ()` ?

**Comment by moosotc** [comment]
Nicolas Ojeda Bar <notifications@github.com> writes:

> Great!  I will fix the font size issues tonight.
>
> @moosotc: what's the best way to multiply all font sizes by the given
> factor `Wsi.fontsizefactor ()` ?
>

all = 2 (reminds me of an old saying - programmers are simpletons they
count - one, two... many)

ui.fontsize (fstate.fontsize) and hint font size (conf.hfsize) (hints =
those things that allow effortless keyboard link navigation when 'F' is
pressed)

-- 
mailto:moosotc@gmail.com

**Comment by nojb** [comment]
Thanks - I will give it a go tonight.

**Comment by nojb** [comment]
@nilsbecker I think everything should be working now, but make sure to delete your `~/.config/llpp.conf` in order for font sizes to get initialized correctly.

**Comment by nilsbecker** [comment]
font sizes seem to work fine now. i have a retina display on the macbook pro and a connected non-retina monitor. both work as expected now. nice.
